### PR TITLE
Change on-device example to use the origin trial API

### PR DESCRIPTION
**Pull request by tomayac**

### CODE REVIEW

**Comment by tomayac** [comment]
Screenshot 2024-10-28 at 18 01 20

**Comment by lozy219** [comment]
I would suggest taking another screenshot after https://chromium-review.googlesource.com/c/chromium/src/+/5969874 is landed, so we can get rid of all the outdated `AIAssistant*` class names.

**Comment by tomayac** [comment]
Makes sense. Note that I was advised to not create a new example, but to rewrite the [existing example](https://github.com/GoogleChrome/chrome-extensions-samples/tree/main/functional-samples/ai.gemini-on-device) to use the origin trial. But I'll be nonetheless sure to wait for the final class names before taking any publicly used screenshots (if need be).

**Comment by tomayac** [comment]
Rewritten the existing demo so it works with the Prompt API extension origin trial.

Screenshot 2024-10-29 at 14 14 22

**Review by sebastianbenz — APPROVED** [review]
Thanks a lot! Are there additional steps required for this to work?

**Comment by sebastianbenz** [comment]
One nit, can you change the error message in sidepanel/index.js to: "Error: chrome.aiOriginTrial not supported in this browser" (or maybe something that makes more sense in the OT context).

**Comment by sebastianbenz** [comment]
Another thing, should we add a minimum Chrome version in the manifest?

**Comment by tomayac** [comment]
Both addressed in https://github.com/GoogleChrome/chrome-extensions-samples/pull/1337/commits/66e809a388c33ee565bb9332f2ef71b8e7e1ec0f. As far as I can tell, once the origin trial is live, this should just work, assuming my token for the pinned version via `key` will be valid for the published CWS version.

File: functional-samples/ai.gemini-on-device/manifest.json [hunk 1]
```diff
@@ -6,7 +6,10 @@
   "background": {
     "service_worker": "background.js"
   },
-  "permissions": ["sidePanel"],
+  "permissions": ["sidePanel", "aiLanguageModelOriginTrial"],
+  "trial_tokens": ["A6hC1yvazc5QTqSESgEo3uwIUs1KBM4VB93yiTdtgFlFlYNJjf18zmzScpaCepaYpNLaMJ4dNkDps+vbrk7oEw8AAAB9eyJvcmlnaW4iOiAiY2hyb21lLWV4dGVuc2lvbjovL2lkZm1qYmJqY21qamtoYWlvZmltbWNtY29qYmlnbmpmIiwgImZlYXR1cmUiOiAiQUlQcm9tcHRBUElGb3JFeHRlbnNpb24iLCAiZXhwaXJ5IjogMTk4OTQwNzUwMH0="],
```
Comment (oliverdunk) [inline functional-samples/ai.gemini-on-device/manifest.json:10]: Thanks! Could we also update the README for this sample to explain how to run Chrome with the right flag for this? We should also mention this is a testing token and explain that developers will be able to register in the future.

Comment (tomayac) [reply functional-samples/ai.gemini-on-device/manifest.json:10]: Dealt with in https://github.com/GoogleChrome/chrome-extensions-samples/pull/1337/commits/68b4923306e46ebbbb71cdeb8974db6018c433f7.

File: functional-samples/ai.gemini-on-device/README.md [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
-The extension provides a chat interface using the prompt API with Chrome's built-in Gemini Nano model.
+The extension provides a chat interface using the Prompt API with Chrome's built-in Gemini Nano model.
 
 ## Running this extension
 
 1. Clone this repository.
-2. Load this directory in Chrome as an [unpacked extension](https://developer.chrome.com/docs/extensions/mv3/getstarted/development-basics/#load-unpacked).
-3. Click the extension icon.
-4. Interact with the prompt API in the sidebar.
+1. Launch Chrome with the following flags:
```
Comment (oliverdunk) [inline functional-samples/ai.gemini-on-device/README.md:12]: `--enable-features=EnablePromptAPIForExtension` isn't required, right? I'm fairly sure the origin trial enables this for you.

**Review by oliverdunk — CHANGES_REQUESTED** [review]

File: functional-samples/ai.gemini-on-device/README.md (hunk 2 above)
Comment (tomayac) [reply functional-samples/ai.gemini-on-device/README.md]: Ah, of course. PTAL.

File: functional-samples/ai.gemini-on-device/README.md [hunk 3]
```diff
@@ -9,12 +9,12 @@ The extension provides a chat interface using the Prompt API with Chrome's built
 ## Running this extension
 
 1. Clone this repository.
-1. Launch Chrome with the following flags:
-   `--enable-features=EnablePromptAPIForExtension --origin-trial-public-key=dRCs+TocuKkocNKa0AtZ4awrt9XKH2SQCI6o4FY6BNA=`
+1. Launch Chrome with the following flag, which makes sure the origin trial token in `manifest.json` is accepted by the browser:
```
Comment (sebastianbenz) [inline functional-samples/ai.gemini-on-device/README.md:12]: Should we mention here that this will go away once the OT is live?

Comment (tomayac) [reply functional-samples/ai.gemini-on-device/README.md]: Good point. Done.

**Comment by sebastianbenz** [comment]
Thanks a lot Tom! This is ready to merge from my POV!

**Pull request merged**